week2/starter/ColorPicker/Views/ContentView.swift

```diff
@@ -0,0 +1,111 @@
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+///
+/// Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+/// distribute, sublicense, create a derivative work, and/or sell copies of the
+/// Software in any work that is designed, intended, or marketed for pedagogical or
+/// instructional purposes related to programming, coding, application development,
+/// or information technology.  Permission for such use, copying, modification,
+/// merger, publication, distribution, sublicensing, creation of derivative works,
+/// or sale is expressly withheld.
+///
+/// This project and source code may use libraries or frameworks that are
+/// released under various Open-Source licenses. Use of those libraries and
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import SwiftUI
+
+struct ContentView: View {
+  @State private var alertIsVisible: Bool = false
+  @State private var redColor: Double = Constants.Genral.initColorVal
+  @State private var greenColor: Double = Constants.Genral.initColorVal
+  @State private var blueColor: Double = Constants.Genral.initColorVal
+  @State private var foregroundColor = Color(red: Constants.Genral.initColorVal, green: Constants.Genral.initColorVal, blue: Constants.Genral.initColorVal)
+  @State private var isEditing = false
+
+  var body: some View {
+
+    VStack {
+      Text("Color Picker")
+        .font(.largeTitle)
+        .foregroundColor(Color("TextColor"))
+
+      RoundedRectangle(cornerRadius: Constants.Genral.colorPickerCornerRadius)
+        .foregroundColor(foregroundColor)
+        .border(.black)
+      SliderView(lebelText: "Red", accentCol: Color.red, sliderColor: $redColor)
+      SliderView(lebelText: "Green", accentCol: Color.green, sliderColor: $greenColor)
+      SliderView(lebelText: "Blue", accentCol: Color.blue,sliderColor: $blueColor)
+      SetColorButton(redColor: redColor, greenColor: greenColor, blueColor: blueColor, foregroundColor: $foregroundColor)
+    }
+    .background(Color("BackgroundColor"))
+    .padding(Constants.Genral.padding)
+
+  }
+
+}
+
+struct SliderView: View {
+  var lebelText: String
+  var accentCol: Color
```

Comment:
This works ok, but it's error prone. Imagine this, every other developer that uses your code must make sure that the name and accent color actually match. There's nothing stopping them from passing name: "Green" and accentColor: .blue.

You can use an enum to make this more safe, for example. This enum would define the colors this slider support (in this case just the RGB colors) and it would have a name and accent color property. So the client of this view would only be able to pass either .red, .blue or .green in the init of the view.